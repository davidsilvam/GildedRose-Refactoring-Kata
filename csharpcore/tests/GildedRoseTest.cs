using Xunit;
using System.Collections.Generic;
using Xunit.Abstractions;//added to debug

using ApprovalTests;
using ApprovalTests.Reporters;

using System.Text;

namespace csharpcore
{
    [UseReporter(typeof(DiffReporter))]
    public class GildedRoseTest
    {
        private readonly ITestOutputHelper _output;

        public GildedRoseTest(ITestOutputHelper output)
        {
            _output = output;
        }

        [Fact]
        public void foo()
        {
            IList<Item> Items = new List<Item> { new Item { Name = "foo", SellIn = 0, Quality = 0 } };
            GildedRose app = new GildedRose(Items);
            app.UpdateQuality();
            //  _output.WriteLine("Debug");
            Assert.Equal("foo", Items[0].Name);
        }

        [Fact]
        public void NormalItemIncreaseAndDecrease()
        {
            IList<Item> Items = new List<Item> {
                new Item {
                    Name = "+5 Dexterity Vest",
                    SellIn = 10,
                    Quality = 20
                },
                new Item {
                    Name = "Elixir of the Mongoose",
                    SellIn = 5,
                    Quality = 7
                }
            };

            GildedRose app = new GildedRose(Items);

            var fakeoutput = new StringBuilder();

            for (var i = 0; i < 31; i++)
            {
                fakeoutput.Append("-------- day " + i + " --------" + "\n");
                for (var j = 0; j < Items.Count; j++)
                {
                    fakeoutput.Append(Items[j].Name + ", " + Items[j].SellIn + ", " + Items[j].Quality + "\n");
                }
                app.UpdateQuality();
            }

            Approvals.Verify(fakeoutput);
        }

        [Fact]
        public void AgedBrieItemIncreaseAndDecrease()
        {
            IList<Item> Items = new List<Item> {
                new Item {
                    Name = "Aged Brie",
                    SellIn = 2,
                    Quality = 0
                }
            };

            GildedRose app = new GildedRose(Items);

            var fakeoutput = new StringBuilder();

            for (var i = 0; i < 31; i++)
            {
                fakeoutput.Append("-------- day " + i + " --------" + "\n");
                for (var j = 0; j < Items.Count; j++)
                {
                    fakeoutput.Append(Items[j].Name + ", " + Items[j].SellIn + ", " + Items[j].Quality + "\n");
                }
                app.UpdateQuality();
            }

            Approvals.Verify(fakeoutput);
        }

        [Fact]
        public void SulfurasItemIncreaseAndDecrease()
        {
            IList<Item> Items = new List<Item> {
                new Item {
                    Name = "Sulfuras, Hand of Ragnaros",
                    SellIn = 0,
                    Quality = 80
                },
                new Item {
                    Name = "Sulfuras, Hand of Ragnaros",
                    SellIn = -1,
                    Quality = 80
                }
            };

            GildedRose app = new GildedRose(Items);

            var fakeoutput = new StringBuilder();

            for (var i = 0; i < 31; i++)
            {
                fakeoutput.Append("-------- day " + i + " --------" + "\n");
                for (var j = 0; j < Items.Count; j++)
                {
                    fakeoutput.Append(Items[j].Name + ", " + Items[j].SellIn + ", " + Items[j].Quality + "\n");
                }
                app.UpdateQuality();
            }

            Approvals.Verify(fakeoutput);
        }

        [Fact]
        public void BackstagePassesItemIncreaseAndDecrease()
        {
            IList<Item> Items = new List<Item> {
               new Item
                {
                    Name = "Backstage passes to a TAFKAL80ETC concert",
                    SellIn = 15,
                    Quality = 20
                },
                new Item
                {
                    Name = "Backstage passes to a TAFKAL80ETC concert",
                    SellIn = 10,
                    Quality = 49
                },
                new Item
                {
                    Name = "Backstage passes to a TAFKAL80ETC concert",
                    SellIn = 5,
                    Quality = 49
                }
            };

            GildedRose app = new GildedRose(Items);

            var fakeoutput = new StringBuilder();

            for (var i = 0; i < 31; i++)
            {
                fakeoutput.Append("-------- day " + i + " --------" + "\n");
                for (var j = 0; j < Items.Count; j++)
                {
                    fakeoutput.Append(Items[j].Name + ", " + Items[j].SellIn + ", " + Items[j].Quality + "\n");
                }
                app.UpdateQuality();
            }

            Approvals.Verify(fakeoutput);
        }

        [Fact]
        public void ConjuredItemIncreaseAndDecrease()
        {
            IList<Item> Items = new List<Item> {
               new Item {
                   Name = "Conjured Mana Cake",
                   SellIn = 3,
                   Quality = 6
                }
            };

            GildedRose app = new GildedRose(Items);

            var fakeoutput = new StringBuilder();

            for (var i = 0; i < 31; i++)
            {
                fakeoutput.Append("-------- day " + i + " --------" + "\n");
                for (var j = 0; j < Items.Count; j++)
                {
                    fakeoutput.Append(Items[j].Name + ", " + Items[j].SellIn + ", " + Items[j].Quality + "\n");
                }
                app.UpdateQuality();
            }

            Approvals.Verify(fakeoutput);
        }
    }
}