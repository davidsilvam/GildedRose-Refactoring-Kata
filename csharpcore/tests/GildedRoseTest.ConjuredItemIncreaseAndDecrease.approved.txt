-------- day 0 --------
Conjured Mana Cake, 3, 6
-------- day 1 --------
Conjured Mana Cake, 2, 4
-------- day 2 --------
Conjured Mana Cake, 1, 2
-------- day 3 --------
Conjured Mana Cake, 0, 0
-------- day 4 --------
Conjured Mana Cake, -1, 0
-------- day 5 --------
Conjured Mana Cake, -2, 0
-------- day 6 --------
Conjured Mana Cake, -3, 0
-------- day 7 --------
Conjured Mana Cake, -4, 0
-------- day 8 --------
Conjured Mana Cake, -5, 0
-------- day 9 --------
Conjured Mana Cake, -6, 0
-------- day 10 --------
Conjured Mana Cake, -7, 0
-------- day 11 --------
Conjured Mana Cake, -8, 0
-------- day 12 --------
Conjured Mana Cake, -9, 0
-------- day 13 --------
Conjured Mana Cake, -10, 0
-------- day 14 --------
Conjured Mana Cake, -11, 0
-------- day 15 --------
Conjured Mana Cake, -12, 0
-------- day 16 --------
Conjured Mana Cake, -13, 0
-------- day 17 --------
Conjured Mana Cake, -14, 0
-------- day 18 --------
Conjured Mana Cake, -15, 0
-------- day 19 --------
Conjured Mana Cake, -16, 0
-------- day 20 --------
Conjured Mana Cake, -17, 0
-------- day 21 --------
Conjured Mana Cake, -18, 0
-------- day 22 --------
Conjured Mana Cake, -19, 0
-------- day 23 --------
Conjured Mana Cake, -20, 0
-------- day 24 --------
Conjured Mana Cake, -21, 0
-------- day 25 --------
Conjured Mana Cake, -22, 0
-------- day 26 --------
Conjured Mana Cake, -23, 0
-------- day 27 --------
Conjured Mana Cake, -24, 0
-------- day 28 --------
Conjured Mana Cake, -25, 0
-------- day 29 --------
Conjured Mana Cake, -26, 0
-------- day 30 --------
Conjured Mana Cake, -27, 0
